### PR TITLE
fix: Clean up required baseline parameters for consistency, cleanup of parameter passing to child process

### LINKED ISSUE #91: bug: fatal error while running default compare
Working on the `main` branch (commit: 7d6f5f926367ebde06dc33a7561c3874d5424a8f), and running through the documentation and trying out the various examples to double-check them, I found that a default baseline compare was failing:

reproduction:
```bash
swift package --allow-writing-to-package-directory benchmark baseline update
swift package benchmark baseline compare
```

Output of failure:
```bash
Building for debugging...
Build complete! (0.21s)
Building benchmark targets in release mode for benchmark run...
Building HistogramBenchmark
Building BenchmarkDateTime
Building Basic
Build complete!
Swift/RangeReplaceableCollection.swift:870: Fatal error: Can't remove last element from an empty collection
error: plugin process ended by an uncaught signal: 5 <command: /usr/bin/sandbox-exec -p '(version 1)
(deny default)
(import "system.sb")
(allow file-read*)
(allow process*)
(allow file-write*
    (subpath "/private/tmp")
    (subpath "/private/var/folders/8t/k6nw7pyx2qq77g8qq_g429080000gn/T")
)
(deny file-write*
    (subpath "/Users/heckj/src/package-benchmark")
)
(allow file-write*
    (subpath "/Users/heckj/src/package-benchmark/.build/plugins/Benchmark-Plugin/outputs")
    (subpath "/Users/heckj/src/package-benchmark/.build/plugins/Benchmark-Plugin/cache")
)
' /Users/heckj/src/package-benchmark/.build/plugins/Benchmark-Plugin/cache/Benchmark_Plugin>, <output:
'Building benchmark targets in release mode for benchmark run...
Building HistogramBenchmark
Building BenchmarkDateTime
Building Basic
Build complete!
Swift/RangeReplaceableCollection.swift:870: Fatal error: Can'\''t remove last element from an empty collection
'>
```

Workaround: if you name the baseline, there's no issue - so running `swift package benchmark baseline compare default` works.

### PR DESCRIPTION
**Pull request by hassila**
## Description

Fixes #91 

## How Has This Been Tested?

Manual testing.

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hassila** [comment]
Current dependencies on/for this PR:
* main
  * **PR #96** <a href="https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96" target="_blank">Graphite</a>  👈
    * **PR #97** <a href="https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96?utm_source=stack-comment).

**Review by dimlio — APPROVED** [review]

File: Plugins/BenchmarkTool/BenchmarkTool.swift [hunk 1]
```diff
@@ -160,6 +142,14 @@ struct BenchmarkTool: AsyncParsableCommand {
                 }
             }
 
+            // Filter out any results that doesn't match regex filtering
```
[the remark this answers is not in the document]
Comment (hassila) [reply Plugins/BenchmarkTool/BenchmarkTool.swift:145]: Yes, it's cleaned up in another PR in the stack.

**Comment by hassila** [comment]
@hassila queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96).

**Comment by hassila** [comment]
[Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96) couldn't merge this PR because it failed optional checks and "ignore optional checks" was not selected.

**Comment by hassila** [comment]
@hassila queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96).

**Comment by hassila** [comment]
@hassila merged this pull request with [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96).

**Pull request merged**